src/Kestrel.Core/Internal/HttpsConnectionAdapter.cs

```diff
@@ -137,7 +159,12 @@ public Task<IAdaptedConnection> OnConnectionAsync(ConnectionAdapterContext conte
 
                 await sslStream.AuthenticateAsServerAsync(sslOptions, CancellationToken.None);
 #else
-                await sslStream.AuthenticateAsServerAsync(_serverCertificate, certificateRequired,
+                var serverCert = _serverCertificate;
+                if (_serverCertificateSelector != null)
+                {
+                    serverCert = _serverCertificateSelector(null, null);
```

Comment:
Do you think we should null check serverCert to get better errors when ServerCertificateSelector is not null but its return value is?